ci/playbooks/tcib/tcib.yml

```diff
@@ -34,32 +34,44 @@
       ansible.builtin.include_role:
         name: build_openstack_packages
 
-    - name: Deploy local registry
-      tags:
-        - container_img_build
-      ansible.builtin.include_role:
-        name: registry_deploy
-
-    - name: Set fact for cifmw_build_containers_image_tag var
-      tags:
-        - container_img_build
+    - name: Construct project change list
       ansible.builtin.set_fact:
-        cifmw_build_containers_image_tag: "{{ cifmw_repo_setup_full_hash }}"
+        zuul_change_list: "{{ zuul_change_list | default([]) + [item.project.short_name] }}"
+        cacheable: true
+      with_items: "{{ zuul['items'] }}"
+      when:
+        - zuul is defined
+        - "'change_url' in item"
 
-    - name: Build containers
-      tags:
-        - container_img_build
-      ansible.builtin.include_role:
-        name: build_containers
+    - name: Build containers when needed
+      when: "'os-net-config' not in zuul_change_list"
+      block:
+        - name: Deploy local registry
+          tags:
+            - container_img_build
+          ansible.builtin.include_role:
+            name: registry_deploy
 
-    - name: Get the containers list from container registry
-      ansible.builtin.uri:
-        url: "http://{{ cifmw_rp_registry_ip }}:5001/v2/_catalog"
-        return_content: true
-      register: cp_imgs
+        - name: Set fact for cifmw_build_containers_image_tag var
+          tags:
+            - container_img_build
+          ansible.builtin.set_fact:
+            cifmw_build_containers_image_tag: "{{ cifmw_repo_setup_full_hash }}"
 
-    - name: Add the container list to file
-      ansible.builtin.copy:
-        content: "{{ cp_imgs.content }}"
-        dest: "{{ ansible_user_dir }}/local_registry.log"
-        mode: "0644"
+        - name: Build containers
+          tags:
+            - container_img_build
+          ansible.builtin.include_role:
+            name: build_containers
+
+        - name: Get the containers list from container registry
+          ansible.builtin.uri:
+            url: "http://{{ cifmw_rp_registry_ip }}:5001/v2/_catalog"
```

Comment:
may as well use the default for the port https://github.com/openstack-k8s-operators/ci-framework/blob/0f6a0c1eba4e34660777065cff91a6e69c71e602/roles/registry_deploy/defaults/main.yml#L24C1-L24C23

Reply:
Done!